Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.10.2-slim-buster as base
+COPY . /opt/
+WORKDIR /opt
+RUN apt-get update && apt-get install curl -y
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+ENV PATH=$PATH:/root/.poetry/bin
+RUN poetry install
```

Comment:
Your Dockerfile works but could be optimised to improve the time to rebuild. Your first layer is `COPY . /opt/`.

That means every time you rebuild your image, if any file at all has changed, the build starts from the beginning.

This means you will waste time reinstalling curl, poetry etc. 

It would be nicer to do something like this order of events:
- install tools
- COPY in your poetry-related files
- RUN poetry install
- COPY in your other files

The result is:
- You never have to reinstall curl/poetry
- If you modify your set of dependencies, then you will redo "poetry install"
- If you modify your app code, you can skip poetry install and just redo that last copy command